packages/venia-ui/lib/components/Wishlist/WishlistDialog/NewWishlistForm/newWishlistForm.js

```diff
@@ -0,0 +1,130 @@
+    } = talonProps;
+
+    const privateRadioText = formatMessage({
+        id: 'newWishlistForm.privateRadio',
+        defaultMessage: 'Private'
+    });
+
+    const publicRadioText = formatMessage({
+        id: 'newWishlistForm.publicRadio',
+        defaultMessage: 'Public'
+    });
+
+    const createButtonText = formatMessage({
+        id: 'newWishlistForm.createButton',
+        defaultMessage: '+ Create a new list'
+    });
+
+    const cancelButtonText = formatMessage({
+        id: 'newWishlistForm.createButton',
+        defaultMessage: 'Cancel'
+    });
+
+    const saveButtonText = formatMessage({
+        id: 'newWishlistForm.saveButton',
+        defaultMessage: 'Save'
+    });
+
+    const maybeForm = isOpen ? (
+        <Fragment>
+            <FormError
+                // classes={{
+                //     root: classes.formErrors
+                // }}
+                errors={formErrors}
+            />
+            <div className={classes.listname}>
+                <Field label="List Name">
+                    <TextInput
+                        id={classes.listname}
+                        field="listname"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+            <div className={classes.visibility}>
+                <RadioGroup
+                    // classes={{
+                    //     radioLabel: classes.radioContents,
+                    //     root: classes.radioRoot
+                    // }}
+                    field="visibility"
+                    initialValue={'PRIVATE'}
+                    items={[
+                        {
+                            label: <span>{privateRadioText}</span>,
+                            value: 'PRIVATE'
+                        },
+                        {
+                            label: <span>{publicRadioText}</span>,
+                            value: 'PUBLIC'
+                        }
+                    ]}
+                />
+            </div>
+            <Button
+                // classes={{
+                //     root_lowPriority: classes.cancelButton
+                // }}
+                onClick={handleCancel}
+                priority="low"
+                type="reset"
+            >
+                {cancelButtonText}
+            </Button>
+            <Button
+                // classes={{
+                //     root_highPriority: classes.confirmButton
+                // }}
+                disabled={isSaveDisabled}
+                onClick={handleSave}
+                priority="high"
+                type="button"
+            >
+                {saveButtonText}
+            </Button>
+        </Fragment>
+    ) : null;
+
+    return (
+        <Fragment>
+            <button onClick={handleNewListClick} type="button">
+                {createButtonText}
+            </button>
+            <Relevant when={() => !!isOpen}>{maybeForm}</Relevant>
```

Comment:
Seems like the right place to use `Relevant`. Should we memoize the `when` function here?

```jsx
const shouldRenderForm = useCallback(
  () => !!isOpen,
  [isOpen]
)
```

Reply:
Done.